warehouse/archive/archiver.go

```diff
@@ -189,6 +192,57 @@ func (*Archiver) usedRudderStorage(metadata []byte) bool {
 
 func (a *Archiver) Do(ctx context.Context) error {
 	a.log.Infof(`[Archiver]: Started archiving for warehouse`)
+
+	uploadsToArchive, err := a.countUploadsToArchive(ctx)
+	if err != nil {
+		return fmt.Errorf("counting uploads to archive: %w", err)
+	}
+
+	for uploadsToArchive > 0 {
+		if err := a.archiveUploads(ctx); err != nil {
+			return fmt.Errorf("archiving uploads: %w", err)
+		}
+		uploadsToArchive -= a.config.maxLimit.Load()
```

Comment:
```suggestion
    maxArchiveLimit := a.config.maxLimit.Load()
	for uploadsToArchive > 0 {
		if err := a.archiveUploads(ctx, maxArchiveLimit); err != nil {
			return fmt.Errorf("archiving uploads: %w", err)
		}
		uploadsToArchive -= maxArchiveLimit
```

Would it be better if we load config only once and pass it everywhere? There can be some issues when we update the config and `archiveUploads` method has a value of `maxLimit = 10000` and we update just before we do `uploadsToArchive -= a.config.maxLimit.Load()`

Reply:
It should be fine since we would rarely change the limit. Also, it should be fine if there are some uploads left because they will get picked up in the next archival.

Reply:
Even if it is fine IMO we should load this only once before the loop for correctness of the code. Its optional though as long as we are able to archive everything we are good.

Reply:
Don't think loading the config would have that kind of performance issue. But in any way I have addressed the comment.